incident_viewer_demo/incident/incident.go

```diff
@@ -65,74 +64,46 @@ func (i *IncidentData) Init(
-	i.BadPeriodEnd = badTimeEnd
 	i.Severity = severity
 	i.NumTestsAffected = testsAffected
 	i.GoodPeriodInfo = goodPeriodInfo
 	i.BadPeriodInfo = badPeriodInfo
 	i.IncidentInfo = incidentInfo
 }
 
-func (i *DefaultIncident) Init(goodTimeStart time.Time, goodTimeEnd time.Time,
-	badTimeStart time.Time,
-	badTimeEnd time.Time,
-	avgDSGood float64,
-	avgDSBad float64,
+/* Assign data members and set appropriate text for information fields */
+func (i *DefaultIncident) MakeIncidentData(goodStartTime time.Time, goodEndTime time.Time,
+	badStartTime time.Time,
+	badEndTime time.Time,
+	avgGoodDS float64,
+	avgBadDS float64,
 	severity float64,
 	testsAffected int) {
 
-	i.goodStartTime = goodTimeStart
-	i.goodEndTime = goodTimeEnd
-	i.badStartTime = badTimeStart
-	i.badEndTime = badTimeEnd
-	i.avgGoodDS = avgDSGood
-	i.avgBadDS = avgDSBad
-	i.severityDecimal = severity
+	// Convert strings to be float64 types described by 2 decimal places
+	gds := strconv.FormatFloat(avgGoodDS, 'f', 2, 64)
+	bds := strconv.FormatFloat(avgGoodDS, 'f', 2, 64)
+	s := strconv.FormatFloat(severity*100, 'f', 2, 64)
+	ta := strconv.Itoa(testsAffected)
+
+	i.goodStartTime = goodStartTime
+	i.goodEndTime = goodEndTime
+	i.badStartTime = badStartTime
+	i.badEndTime = badEndTime
+	i.avgGoodDS = avgGoodDS
+	i.avgBadDS = avgBadDS
+	i.severity = severity
 	i.numTestsAffected = testsAffected
-}
-
-//**************************************//
-//    		GETTER METHODS
-//**************************************//
-
-func (i *DefaultIncident) GetGoodPeriod() (time.Time, time.Time) {
-	return i.goodStartTime, i.goodEndTime
-}
-
-func (i *DefaultIncident) GetBadPeriod() (time.Time, time.Time) {
-	return i.badStartTime, i.badEndTime
-}
-
-func (i *DefaultIncident) GetGoodMetric() float64 {
-	return i.avgGoodDS
-}
-
-func (i *DefaultIncident) GetBadMetric() float64 {
-	return i.avgBadDS
-}
-
-func (i *DefaultIncident) GetSeverity() float64 {
-	return i.severityDecimal
-}
-
-func (i *DefaultIncident) GetTestsAffected() int {
-	return i.numTestsAffected
-}
-
-func (i *DefaultIncident) GetGoodPeriodInfo() string {
-	ds := strconv.FormatFloat(i.avgGoodDS, 'f', 2, 64)
-	return "Average download speed: " + ds + " mb/s"
-}
+	i.goodPeriodInfo = "Average download speed: " + gds + " mb/s"
+	i.badPeriodInfo = "Average download speed: " + bds + " mb/s"
+	i.incidentInfo = "Download speed dropped by " + s + "% affecting " + ta + " tests"
 
-func (i *DefaultIncident) GetBadPeriodInfo() string {
-	ds := strconv.FormatFloat(i.avgBadDS, 'f', 2, 64)
-	return "Average download speed: " + ds + " mb/s"
 }
 
-func (i *DefaultIncident) GetIncidentInfo() string {
-	s := strconv.FormatFloat(i.severityDecimal*100, 'f', 2, 64)
-	ta := strconv.Itoa(i.numTestsAffected)
-	return "Download speed dropped by " + s + "% affecting " + ta + " tests"
+/* Retrieve all Incident fields */
+func (i *DefaultIncident) GetIncidentData() (time.Time, time.Time, time.Time, time.Time, float64, float64, float64, int, string, string, string) {
```

Comment:
I like how this new approach simplifies our code. The traditional OG way makes the code more readable though, but we got to go with what Peter wants.